File: test_data/csharp/test_main/aas_core_meta.v3/input/snippets/Verification/data_specification_IEC_61360s_have_value.cs
/// <summary>
/// Check that the <see cref="Aas.IDataSpecificationIec61360.Value" /> is defined
/// for all data specifications whose content is given as IEC 61360.
/// </summary>
[CodeAnalysis.SuppressMessage("ReSharper", "InconsistentNaming")]
public static bool DataSpecificationIec61360sHaveValue(
    IEnumerable<Aas.IEmbeddedDataSpecification> embeddedDataSpecifications
)
{
    foreach (var embeddedDataSpecification in embeddedDataSpecifications)
    {
        var iec61360 = (
            embeddedDataSpecification.DataSpecificationContent
                as Aas.IDataSpecificationIec61360
        );
        if (iec61360 != null)
        {
            if (iec61360.Value == null)
            {
                return false;
            }
        }
    }

    return true;
}

File: test_data/java/test_main/aas_core_meta.v3/input/snippets/Verification/data_specification_IEC_61360s_have_value.java
/** Check that the {@link IDataSpecificationIec61360#getValue() value}  is defined
* for all data specifications whose content is given as IEC 61360.
*/
public static boolean dataSpecificationIec61360sHaveValue(
        Iterable<IEmbeddedDataSpecification> embeddedDataSpecifications
){
    for (IEmbeddedDataSpecification embeddedDataSpecification : embeddedDataSpecifications){
        IDataSpecificationIec61360 iec61360 = (IDataSpecificationIec61360) embeddedDataSpecification.getDataSpecificationContent();
        if (iec61360 != null)
        {
            if (!iec61360.getValue().isPresent())
            {
                return false;
            }
        }
    }
    return true;
}